intel_extension_for_transformers/transformers/utils/quantization_config.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Configs for intel extension for transformers."""
+
+from dataclasses import dataclass
+from typing import Optional, Any
+from transformers import BitsAndBytesConfig
+
+
+@dataclass
+class WeightOnlyQuantizationConfig:
+    algorithm: str = 'RTN'
+    bits: int = 8
+    group_size: int = -1
+    scheme: str = 'sym'
+    enable_full_range: bool = True
+
+@dataclass
+class AMPConfig:
+    dtype: str = 'bfloat16'    
+
+@dataclass
+class SmoothQuantConfig:
+    tokenizer: Any = None
+    calib_func: Any = None

```

Comment:
1. why there is no dataset related setting exposed and hard code in internal? 
2. calib_func is quite ugly. is possible to remove it? FX tracing/ multiple modules issue shouldn't introduce such argument and bring complexity.

Reply:
reply Q1:
for built-in calib_func, I don't know what key for each sample if we open the args.dataset. eg: example["text"], example["code"]
reply Q2:
I think the default calib_func is calib_func is mainly aimed at LLM quantization, for other models such as qa task specifically is not suitable, user need provided calib_func by themselves.